sdc/tests/test_series.py

```diff
@@ -3867,13 +4067,53 @@ def pyfunc(series, periods, freq, axis, fill_value):
         cfunc = self.jit(pyfunc)
         freq = None
         axis = 0
-        for data in test_global_input_data_float64:
-            series = pd.Series(data)
-            for periods in [-2, 0, 3]:
-                for fill_value in [9.1, np.nan, -3.3, None]:
-                    jit_result = cfunc(series, periods, freq, axis, fill_value)
-                    ref_result = pyfunc(series, periods, freq, axis, fill_value)
-                    pd.testing.assert_series_equal(jit_result, ref_result)
+        datas = test_global_input_data_signed_integer64 + test_global_input_data_float64
+        for data in datas:
+            for periods in [1, 2, -1]:
+                for fill_value in [-1, 0, 9.1, np.nan, -3.3, None]:
+                    with self.subTest(data=data, periods=periods, fill_value=fill_value):
+                        series = pd.Series(data)
+                        jit_result = cfunc(series, periods, freq, axis, fill_value)
+                        ref_result = pyfunc(series, periods, freq, axis, fill_value)
+                        pd.testing.assert_series_equal(jit_result, ref_result)
+
+    @sdc_limitation
+    def test_series_shift_0period(self):
+        def pyfunc():
+            series = pd.Series([6, 4, 3])
+            return series.shift(periods=0)
+
+        cfunc = self.jit(pyfunc)
+        ref_result = pyfunc()
+        pd.testing.assert_series_equal(cfunc(), ref_result)
+
+    def test_series_shift_0period_sdc(self):
+        def pyfunc():
+            series = pd.Series([6, 4, 3])
+            return series.shift(periods=0)
+
+        cfunc = self.jit(pyfunc)
+        ref_result = pyfunc()
+        pd.testing.assert_series_equal(cfunc(), ref_result.astype(np.float64))
```

Comment:
Not sure if that's really better, but I think it's good to have a comment in tests where we attempt to modify ref_result in some way or use less strict checks.
```suggestion
        # check_dtype=False due to SDC impl returns fixed dtype - float64
        pd.testing.assert_series_equal(cfunc(), ref_result, check_dtype=False)
```

Reply:
done